packages-exp/messaging-exp/src/helpers/register.ts

```diff
@@ -22,26 +22,42 @@ import {
   InstanceFactory
 } from '@firebase/component';
 import { ERROR_FACTORY, ErrorCode } from '../util/errors';
+import { isSwSupported, isWindowSupported } from '../api/isSupported';
 
 import { MessagingService } from '../messaging-service';
 import { _registerComponent } from '@firebase/app-exp';
-import { isSupported } from './isSupported';
 
 const messagingFactory: InstanceFactory<'messaging-exp'> = (
   container: ComponentContainer
 ) => {
-  if (!isSupported()) {
-    throw ERROR_FACTORY.create(ErrorCode.UNSUPPORTED_BROWSER);
-  }
-
   return new MessagingService(
     container.getProvider('app-exp').getImmediate(),
     container.getProvider('installations-exp-internal').getImmediate(),
     container.getProvider('analytics-internal')
   );
 };
 
-export function registerMessaging(): void {
+export function registerWindowMessaging(): void {
+  // Top-level 'await' requires 'module' option set to 'esnext' or 'system', and 'target' option
+  // set to 'es2017' or higher. For compatibility, use async expression here.
+  void (async () => {
```

Comment:
This unfortunately still doesn't work because `registerMessaging()` gets executed automatically on import. We should move the check into `messagingFactory`

Reply:
couple reasons:
to match the other isSuppoorted API (auth?)
and be able to do idb.openDb check

Reply:
copy. done